### PR TITLE
AC - Modal Inventory

### PR DESCRIPTION
**Pull request by KevinDurand974**
WIP

- [x] Waiting for the 3 fixed size for modal
- [x] Change all defined size on all modals
- [x] Waiting check, if modals size are good

From issue: https://github.com/blockframes/blockframes/issues/6507

### CODE REVIEW

File: libs/contract/src/lib/term/components/detailed/mediaToRights.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { CommonModule } from '@angular/common';
+
+@Pipe({
+  name: 'mediaToRights'
+})
+export class MediaToRightsPipe implements PipeTransform {
+  transform(value: string): string {
+    return value === 'medias' ? 'rights' : value;
+  }
+}
```
Comment (phpgeekfr) [inline libs/contract/src/lib/term/components/detailed/mediaToRights.pipe.ts:11]: it seems a bit overkill to have a pipe just for this. can't you do without it ?

```
{{ (data.scope === 'medias' ? 'rights' : data.scope ) | titlecase }}
```

File: libs/ui/src/lib/global-modal/global-modal.component.ts [hunk 2]
```diff
@@ -20,4 +30,17 @@ export class GlobalModalComponent {
...
+    switch(this.data?.style) {
+      case 'large':
+        return 'sizeL';
+      case 'medium':
+        return 'sizeM';
+      case 'small':
+        return 'sizeS';
+      default:
+        return 'sizeM';
+    }
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.ts:44]: why not pass the right value directly ?

File: libs/event/src/lib/form/create/create.component.html [hunk 3]
```diff
@@ -1,25 +1,24 @@
-﻿<button mat-icon-button class="mat-dialog-close" mat-dialog-close>
-  <mat-icon svgIcon="close"></mat-icon>
-</button>
+﻿<global-modal>
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/create/create.component.html:1]: form is not displayed entirely 

image

File: libs/event/src/lib/form/shell/shell.component.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Component, ChangeDetectionStrategy, OnInit, ViewChild, TemplateRef, ChangeDetectorRef, OnDestroy, Inject } from '@angular/core';
+import { Component, ChangeDetectionStrategy, OnInit, ChangeDetectorRef, OnDestroy, Inject } from '@angular/core';
 import { Router, ActivatedRoute, RouterOutlet } from '@angular/router';
 import { routeAnimation } from '@blockframes/utils/animations/router-animations';
```
Comment (phpgeekfr) [inline libs/event/src/lib/form/shell/shell.component.ts:3]: image

image

padding is not the same (cross position, space after last button)

File: apps/catalog/catalog/src/app/dashboard/sales/shell/negotiation/negotiation.component.ts [hunk 5]
```diff
@@ -84,12 +96,13 @@ export class NegotiationComponent implements NegotiationGuardedComponent, OnInit
       this.snackBar.open('Your counter offer has been sent', null, config);
       this.router.navigate(['..', 'view'], { relativeTo: this.route });
     };
-    const data = {
+    const data: Data = {
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/dashboard/sales/shell/negotiation/negotiation.component.ts:99]: sometimes, data passed to component is typed, sometimes it is not.

on `libs/ui/src/lib/global-modal/global-modal.component.ts` you can do: 
```
type Size =  'large' | 'medium' | 'small';
interface CommonModalData {
  size: Size,
  selectedFiles?: StorageFile[],
  onClose?: () => void
}

export function createModalData<T>(data: T, size: Size): T & CommonModalData {
  return {
    ...data,
    size
  }
}
```


example usage:
```
    const data = createModalData<DeclineData>( {
      type: 'seller',
      showAcceptTermsCheckbox: true,
    }, 'small');
```

```
    const data = createModalData( {
      onConfirm,
      title: 'Are you sure to submit this contract?',
      question: 'Please verify if all the contract elements are convenient for you.',
      confirm: 'Yes, submit',
      cancel: 'Come back & verify contract',
    }, 'small');
```

File: libs/ui/src/lib/global-modal/global-modal.component.ts [hunk 6]
```diff
@@ -20,4 +33,9 @@ export class GlobalModalComponent {
     if (this.data?.onClose) return this.data.onClose();
     return this.dialogRef.close(false);
   }
+
+  public setModalType() {
+    if (this.data?.size) return this.data.size;
+    return 'medium';
+  }
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.ts:40]: according to `createModalData`, Size will always be defined.
You should be able to convert to public and use it directly in HTML

`@Inject(MAT_DIALOG_DATA) private public: CommonModalData,`

File: apps/catalog/catalog/src/app/dashboard/sales/shell/external/external.component.ts [hunk 7]
```diff
@@ -20,19 +20,20 @@ export class ExternalSaleComponent {
...
 
-
   changeStatus(status: ContractStatus, id: string) {
     if (status === 'declined') {
-      const data: ConfirmDeclineData = { type: 'seller' };
-      const ref = this.dialog.open(ConfirmDeclineComponent, { data });
+      const ref = this.dialog.open(ConfirmDeclineComponent, {
+        data: createModalData<ConfirmDeclineData>({
+          type: 'seller'
+        }, 'medium')
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/dashboard/sales/shell/external/external.component.ts:35]: `medium` is not needed

File: libs/utils/src/lib/gdpr-cookie/cookie-banner/cookie-banner.component.ts [hunk 8]
```diff
@@ -35,7 +36,9 @@ export class CookieBannerComponent implements OnInit {
 
   /** Opens a dialog with terms of use and privacy policy given by the parent. */
   public openPrivacyPolicy() {
-    this.dialog.open(PrivacyPolicyComponent, { maxHeight: '80vh' })
+    this.dialog.open(PrivacyPolicyComponent, {
+      data: createModalData({}, 'large')
+    });
```
Comment (phpgeekfr) [inline libs/utils/src/lib/gdpr-cookie/cookie-banner/cookie-banner.component.ts:41]:
```suggestion
    this.dialog.open(PrivacyPolicyComponent, { data: createModalData({}, 'large') });
```

File: libs/ui/src/lib/open-preview/open-preview.component.ts [hunk 9]
```diff
@@ -22,8 +23,10 @@ export class OpenPreviewComponent {
 
   async openModal() {
     this.dialogRef = this.dialog.open(PreviewFileModalComponent, {
+      data: createModalData({
+        ...this.ref
+      }, 'medium'),
```
Comment (phpgeekfr) [inline libs/ui/src/lib/open-preview/open-preview.component.ts:28]:
```suggestion
      data: createModalData({ ...this.ref }, 'medium'),
```

File: libs/ui/src/lib/global-modal/global-modal.component.html [hunk 10]
```diff
@@ -1,9 +1,11 @@
...
-  <button mat-icon-button (click)="close()">
-    <mat-icon svgIcon="close"></mat-icon>
-  </button>
-</mat-dialog-header>
-
-<mat-dialog-content class="mat-dialog-content" fxLayout="column">
-  <ng-content></ng-content>
-</mat-dialog-content>
\ No newline at end of file
+<section [class]="setModalType()">
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.html:1]:
```suggestion
<section [class]="data.size">
```

File: libs/organization/src/lib/organization/pages/member/member.component.ts [hunk 11]
```diff
@@ -75,16 +76,16 @@ export class MemberComponent implements OnInit {
...
           question: 'If you remove a member from you company, you will be able to invite this person again.',
           confirm: 'Yes, remove member.',
           cancel: 'No, keep member.',
           onConfirm: async () => {
             await this.orgService.removeMember(uid);
             this.snackBar.open('Member removed.', 'close', { duration: 2000 });
-          },
-        }
+          }
+        }, 'medium')
```
Comment (phpgeekfr) [inline libs/organization/src/lib/organization/pages/member/member.component.ts:88]:
```suggestion
        })
```

File: libs/movie/src/lib/movie/marketplace/shell/shell.component.ts [hunk 12]
```diff
@@ -69,10 +70,11 @@ export class TitleMarketplaceShellComponent implements OnInit {
...
   fullscreen(refs: StorageFile[], index: number) {
     this.dialog.open(FileListPreviewComponent, {
-      data: { refs, index },
-      width: '80vw',
-      height: '80vh',
-      autoFocus: false,
+      data: createModalData({
+        refs,
+        index
+      }, 'large'),
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/marketplace/shell/shell.component.ts:76]:
```suggestion
      data: createModalData({ refs,  index }, 'large'),
```

File: libs/movie/src/lib/movie/form/main/main.component.ts [hunk 13]
```diff
@@ -20,6 +20,7 @@ import { maxYear } from '@blockframes/utils/form/validators/validators';
 import { MatDialog } from '@angular/material/dialog';
 import { Filmography } from '@blockframes/model';
 import { displayFilmographies } from '@blockframes/movie/pipes/filmography.pipe';
+import { createModalData } from '@blockframes/ui/global-modal/global-modal.component';
 @Component({
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/form/main/main.component.ts:24]:
```suggestion
import { createModalData } from '@blockframes/ui/global-modal/global-modal.component';

@Component({
```

File: libs/media/src/lib/file/picker/picker.component.ts [hunk 14]
```diff
@@ -119,10 +120,10 @@ export class FilePickerComponent implements OnInit, OnDestroy {
 
   previewFile(ref: StorageFile) {
     this.dialog.open(FilePreviewComponent, {
-      data: { ref },
-      width: '70vw',
-      height: '70vh',
-      autoFocus: false,
+      data: createModalData({
+        ref
+      }, 'large'),
```
Comment (phpgeekfr) [inline libs/media/src/lib/file/picker/picker.component.ts:125]:
```suggestion
      data: createModalData({ ref }, 'large'),
```

File: libs/media/src/lib/file/carousel/carousel.component.ts [hunk 15]
```diff
@@ -38,12 +39,10 @@ export class FileCarouselComponent {
...
     this.dialog.open(FilePickerComponent, {
-      width: '80%',
-      height: '80%',
-      disableClose: true,
-      data: {
-        selectedFiles: this.files,
-      }
+      data: createModalData({
+        selectedFiles: this.files
+      }, 'large'),
```
Comment (phpgeekfr) [inline libs/media/src/lib/file/carousel/carousel.component.ts:44]:
```suggestion
      data: createModalData({ selectedFiles: this.files }, 'large'),
```

File: libs/event/src/lib/components/week/week.component.ts [hunk 16]
```diff
@@ -134,8 +135,13 @@ export class CalendarWeekComponent {
 
   /** Open a create dialog and redirect if needed */
   private createEvent(calEvent: CalendarEvent) {
-    const data = { event: { ...calEvent, type: '' }, types: this.eventTypes };
-    this.dialog.open(EventCreateComponent, { data, width: '650px', autoFocus: false }).afterClosed()
+    this.dialog.open(EventCreateComponent, {
+      data: createModalData({
+        event: { ...calEvent, type: '' },
+        types: this.eventTypes
+      }, 'medium'),
```
Comment (phpgeekfr) [inline libs/event/src/lib/components/week/week.component.ts:142]:
```suggestion
      }),
```

File: libs/contract/src/lib/negotiation/form/negotiation-form.component.ts [hunk 17]
```diff
@@ -40,8 +41,10 @@ export class NegotiationFormComponent {
 
   openDetails(terms: string, scope: Scope) {
     this.dialog.open(DetailedTermsComponent, {
-      data: { terms, scope },
-      maxHeight: '80vh',
+      data: createModalData({
+        terms,
+        scope
+      }, 'medium'),
```
Comment (phpgeekfr) [inline libs/contract/src/lib/negotiation/form/negotiation-form.component.ts:47]:
```suggestion
      data: createModalData({ terms, scope }),
```

File: libs/contract/src/lib/contract/holdback/selection-modal/selection-modal.component.ts [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+  close() {
+    this.dialogRef.close();
+  }
+
+  openDetails(terms: string, scope: Scope) {
+    this.dialog.open(DetailedTermsComponent, {
+      data: createModalData({
+        terms,
+        scope
+      }, 'medium'),
```
Comment (phpgeekfr) [inline libs/contract/src/lib/contract/holdback/selection-modal/selection-modal.component.ts:50]:
```suggestion
      data: createModalData({ terms, scope }),
```

File: libs/contract/src/lib/contract/components/item/contract-item.component.ts [hunk 19]
```diff
@@ -33,11 +34,21 @@ export class ContractItemComponent {
   ) { }
 
   openDetails(terms: string[], scope: Scope) {
-    this.dialog.open(DetailedTermsComponent, { data: { terms, scope }, maxHeight: '80vh', autoFocus: false });
+    this.dialog.open(DetailedTermsComponent, {
+      data: createModalData({
+        terms,
+        scope
+      }, 'medium'),
```
Comment (phpgeekfr) [inline libs/contract/src/lib/contract/components/item/contract-item.component.ts:41]:
```suggestion
      data: createModalData({ terms, scope }),
```

File: libs/campaign/src/lib/marketplace/financing/financing.component.ts [hunk 20]
```diff
@@ -74,7 +75,7 @@ export class MarketplaceFinancingComponent {
           await this.consentsService.createConsent('access', campaignId, file);
           window.open(file, '_blank');
         }
-      }
+      }, 'medium')
```
Comment (phpgeekfr) [inline libs/campaign/src/lib/marketplace/financing/financing.component.ts:78]:
```suggestion
      })
```

File: apps/festival/festival/src/app/marketplace/title/view/view.component.ts [hunk 21]
```diff
@@ -78,9 +78,7 @@ export class MarketplaceMovieViewComponent {
 
   requestAskingPrice(movieId: string) {
     const ref = this.dialog.open(RequestAskingPriceComponent, {
-      data: { movieId },
-      maxHeight: '80vh',
-      maxWidth: '650px',
+      data: { movieId, style: 'large' },
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/marketplace/title/view/view.component.ts:81]:
```suggestion
      data: createModalData({ movieId },'large' ),
```

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 22]
```diff
@@ -72,15 +73,13 @@ export class ListComponent {
...
-      data: { title, values: displayPerson(values) },
-      maxHeight: '80vh',
-      minWidth: '50vw',
-      maxWidth: '80vw',
-      minHeight: '50vh',
-      autoFocus: false,
+      data: createModalData({
+        title,
+        values: displayPerson(values)
+      }, 'medium'),
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:81]:
```suggestion
      data: createModalData({ title, values: displayPerson(values)}),
```

File: apps/crm/crm/src/app/crm/pages/user/user.component.ts [hunk 23]
```diff
@@ -245,7 +246,13 @@ export class UserComponent implements OnInit {
   }
 
   openDetails(terms: string[], scope: Scope) {
-    this.dialog.open(DetailedTermsComponent, { data: { terms, scope }, maxHeight: '80vh', autoFocus: false });
+    this.dialog.open(DetailedTermsComponent, {
+      data: createModalData({
+        terms,
+        scope
+      }, 'medium'),
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/user/user.component.ts:253]:
```suggestion
      data: createModalData({  terms,  scope  }),
```

File: apps/crm/crm/src/app/crm/pages/user/user.component.ts [hunk 24]
```diff
@@ -208,7 +209,7 @@ export class UserComponent implements OnInit {
           this.snackBar.open('User deleted !', 'close', { duration: 5000 });
           this.router.navigate(['c/o/dashboard/crm/users']);
         }
-      }
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/user/user.component.ts:212]:
```suggestion
      })
```

File: apps/crm/crm/src/app/crm/pages/organizations/organizations.component.ts [hunk 25]
```diff
@@ -52,10 +52,7 @@ export class OrganizationsComponent {
   }
 
   createOrg() {
-    this.dialog.open(OrganizationCreateComponent, {
-      height: '80vh',
-      width: '60vw',
-    });
+    this.dialog.open(OrganizationCreateComponent, { data: { style: 'large' } });
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/organizations/organizations.component.ts:55]: use `createModalData(`

File: apps/crm/crm/src/app/crm/pages/offers/shell/offer-view/offer-view.component.ts [hunk 26]
```diff
@@ -84,34 +85,34 @@ export class OfferViewComponent implements OnDestroy, OnInit {
...
         confirmationWord: 'update',
         placeholder: 'To confirm the update, please write « UPDATE » in the field below.',
         confirm: 'Confirm and update',
         confirmButtonText: 'Confirm and update',
         cancel: 'Cancel',
-        onConfirm,
-      },
-      autoFocus: true,
+        onConfirm
+      }, 'medium'),
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/offers/shell/offer-view/offer-view.component.ts:97]:
```suggestion
      }),
```

File: apps/crm/crm/src/app/crm/pages/offers/shell/contract-view/contract-view.component.ts [hunk 27]
```diff
@@ -76,14 +77,14 @@ export class ContractViewComponent implements OnInit, OnDestroy {
...
+      data: createModalData({
         title: 'Are you sure ?',
         subtitle: `You are about to delete permanently this term (#${term.id}). This action will also update the contract #${term.contractId} to remove the reference to the deleted term.`,
         text: `Please type "DELETE" to confirm.`,
         confirmationWord: 'DELETE',
         confirmButtonText: 'Delete term',
-        onConfirm: this.delete(term),
-      }
+        onConfirm: this.delete(term)
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/offers/shell/contract-view/contract-view.component.ts:87]:
```suggestion
      })
```

File: apps/crm/crm/src/app/crm/pages/external-sales/view/view.component.ts [hunk 28]
```diff
@@ -60,14 +61,14 @@ export class ContractViewComponent {
...
+      data: createModalData({
         title: 'Are you sure ?',
         subtitle: `You are about to delete permanently this term (#${term.id}). This action will also update the contract #${term.contractId} to remove the reference to the deleted term.`,
         text: `Please type "DELETE" to confirm.`,
         confirmationWord: 'DELETE',
         confirmButtonText: 'Delete term',
-        onConfirm: this.delete(term),
-      }
+        onConfirm: this.delete(term)
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/external-sales/view/view.component.ts:71]:
```suggestion
      })
```

File: apps/crm/crm/src/app/crm/pages/movie/movie.component.ts [hunk 29]
```diff
@@ -123,8 +124,8 @@ export class MovieComponent implements OnInit {
           await this.movieService.remove(this.movie.id);
           this.snackBar.open('Movie deleted !', 'close', { duration: 5000 });
           this.router.navigate(['c/o/dashboard/crm/movies']);
-        },
-      },
+        }
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/crm/crm/src/app/crm/pages/movie/movie.component.ts:128]:
```suggestion
      })
```

File: apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.ts [hunk 30]
```diff
@@ -151,7 +151,7 @@ export class MarketplaceSelectionComponent {
         { duration: 2000 }
       );
     } else {
-      this.dialog.open(SpecificTermsComponent, { data: { currency: this.currencyForm.value } });
+      this.dialog.open(SpecificTermsComponent, { data: { currency: this.currencyForm.value, style: 'large' } });
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.ts:154]: use `createModalData(`

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts [hunk 31]
```diff
@@ -175,20 +176,30 @@ export class MarketplaceMovieAvailsComponent implements AfterViewInit, OnDestroy
...
   }
 
   /** Open a modal to display the entire list of territories when this one is too long */
   public openTerritoryModal(term: BucketTerm) {
-    this.dialog.open(DetailedTermsComponent, { data: { terms: term.territories, scope: 'territories' }, maxHeight: '80vh', autoFocus: false });
+    this.dialog.open(DetailedTermsComponent, {
+      data: createModalData({
+        terms: term.territories,
+        scope: 'territories'
+      }, 'medium'),
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:190]: oneline & remove medium

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts [hunk 32]
```diff
@@ -175,20 +176,30 @@ export class MarketplaceMovieAvailsComponent implements AfterViewInit, OnDestroy
...
   }
 
   /** Open a modal to display holdback warnings */
   openHoldbackModal(holdbacks: Holdback[]) {
-    this.dialog.open(HoldbackModalComponent, { data: { holdbacks, withWarning: true }, maxHeight: '80vh' });
+    this.dialog.open(HoldbackModalComponent, {
+      data: createModalData({
+        holdbacks,
+        withWarning: true
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:201]: same

File: apps/catalog/catalog/src/app/marketplace/offer/shell/contract-view/contract-view.component.ts [hunk 33]
```diff
@@ -49,15 +48,14 @@ export class ContractViewComponent {
...
-    };
-    this.dialog.open(ConfirmWithValidationComponent, { data });
+    this.dialog.open(ConfirmWithValidationComponent, {
+      data: createModalData({
+        onConfirm,
+        title: 'Are you sure you wish to accept this contract?',
+        question: 'Please verify if all the contract elements are convenient for you.',
+        confirm: 'Yes, accept Contract',
+        cancel: 'Come back & Verify Contract'
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/marketplace/offer/shell/contract-view/contract-view.component.ts:58]:
```suggestion
      })
```

File: apps/catalog/catalog/src/app/dashboard/title/list/list.component.ts [hunk 34]
```diff
@@ -76,15 +77,13 @@ export class TitleListComponent {
...
-      data: { title, values: displayPerson(values) },
-      maxHeight: '80vh',
-      minWidth: '50vw',
-      maxWidth: '80vw',
-      minHeight: '50vh',
-      autoFocus: false,
+      data: createModalData({
+        title,
+        values: displayPerson(values)
+      }, 'medium'),
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/dashboard/title/list/list.component.ts:85]: oneline & remove medium

File: apps/catalog/catalog/src/app/dashboard/sales/shell/view/view.component.ts [hunk 35]
```diff
@@ -32,14 +33,15 @@ export class SaleViewComponent {
...
-    };
-    const ref = this.dialog.open(ConfirmWithValidationComponent, { data });
+    const ref = this.dialog.open(ConfirmWithValidationComponent, {
+      data: createModalData({
+        onConfirm: () => this.negotiationService.update(negotiationId, { status }, options),
+        title: 'Are you sure you want to accept this Contract?',
+        question: 'Please verify if all the contract elements are convenient for you.',
+        confirm: 'Yes, accept Contract',
+        cancel: 'Come back & verify Contract'
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/dashboard/sales/shell/view/view.component.ts:43]:
```suggestion
      })
```

File: apps/catalog/catalog/src/app/dashboard/sales/shell/negotiation/negotiation.component.ts [hunk 36]
```diff
@@ -54,11 +55,12 @@ export class NegotiationComponent implements NegotiationGuardedComponent, OnInit
...
-    const data: ConfirmDeclineData = { 
-      type: 'seller',
-      showAcceptTermsCheckbox: true 
-    };
-    const ref = this.dialog.open(ConfirmDeclineComponent, { data });
+    const ref = this.dialog.open(ConfirmDeclineComponent, {
+      data: createModalData<ConfirmDeclineData>({
+        type: 'seller',
+        showAcceptTermsCheckbox: true
+      }, 'medium')
```
Comment (phpgeekfr) [inline apps/catalog/catalog/src/app/dashboard/sales/shell/negotiation/negotiation.component.ts:62]: oneline & remove medium

File: libs/ui/src/lib/global-modal/global-modal.component.ts [hunk 37]
```diff
@@ -1,17 +1,30 @@
...
+  return { ...data, size };
+}
+
 @Component({
   selector: 'global-modal',
   templateUrl: './global-modal.component.html',
   styleUrls: ['./global-modal.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.ts:22]: why do you need this ?

File: libs/ui/src/lib/global-modal/global-modal.component.scss [hunk 38]
```diff
@@ -1,3 +1,50 @@
 :host {
   display: block;
 }
+
+section {
+  position: relative;
+
+  .closeButton {
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.scss:8]: `header` is not enough ?

File: libs/ui/src/lib/global-modal/global-modal.component.ts (hunk 37 above)
Comment (KevinDurand974) [reply libs/ui/src/lib/global-modal/global-modal.component.ts:22]: I dont need this, it was for a test, to change the css but i dont use it anymore, i forgot to remove it

File: libs/ui/src/lib/global-modal/global-modal.component.scss (hunk 38 above)
Comment (KevinDurand974) [reply libs/ui/src/lib/global-modal/global-modal.component.scss:8]: I don't know why I did that, so i removed the header and put the class on the button 🤷‍♂️

File: libs/ui/src/lib/global-modal/global-modal.component.ts [hunk 39]
```diff
@@ -18,8 +18,7 @@ export function createModalData<T>(data: T, size: Size = 'medium'): T & CommonMo
   selector: 'global-modal',
   templateUrl: './global-modal.component.html',
   styleUrls: ['./global-modal.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None
+  changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (phpgeekfr) [inline libs/ui/src/lib/global-modal/global-modal.component.ts:21]: Remove also the import `ViewEncapsulation`

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**